prac_06/programming_language.py

```diff
@@ -0,0 +1,21 @@
+"""Class defining a Programming Language."""
+
+
+class ProgrammingLanguage:
+    """Represents a Programming Language object."""
+
+    def __init__(self, name, typing, reflection, year_created):
+        """Initialise a Programming Language instance."""
+        self.name = name
+        self.typing = typing
+        self.reflection = bool(reflection)
```

Comment:
You could simplify this line by writing it as self.reflection = reflection.

Reply:
This is true, but that is because we only created a function for the typing data. If we wanted to add another function related to reflection or year_created, they need to be stored as their respective types (Boolean and Integer). This is assuming all input is evil and based on the langauges.py program, where there is no declaration of bool() or int() when creating the objects.